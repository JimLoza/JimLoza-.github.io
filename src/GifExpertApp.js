import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = []}) => {

    // const categorias = ['One Punch', 'Samurai X', 'Dragon Ball'];
    // const [categorias, setCategorias] = useState(['One Punch']);

    const [categorias, setCategorias] = useState( defaultCategories );

    // const handleadd = () => {
    //     // setCategorias( [...categorias, 'HunterXHunter'] );
    //     setCategorias( cats =>  [...cats, 'HunterXHunter'] );
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategorias={ setCategorias }/>
            <hr />

            <ol>
                { 
                    categorias.map( category => ( //Estamos haciendo obvio que hay un return encerrando entre parentesis el GifGrid 
                        <GifGrid 
                            key={ category }
                            category={ category } 
                        />
                    ))
                }
            </ol>

        </>
    )
}
