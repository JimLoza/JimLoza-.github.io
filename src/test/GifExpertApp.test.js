import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe('Haciendo pruebas en componente <GifExpertApp/>', () => {
    

    test('Tiene que hacer match con el snap', () => {
        
        const wrapper = shallow( <GifExpertApp/> );

        expect( wrapper ).toMatchSnapshot();
    });


    test('Debe de mostrar yba lista de categorias', () => {
        
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
    
    
})
