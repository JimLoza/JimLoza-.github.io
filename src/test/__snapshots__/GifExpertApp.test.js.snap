// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Haciendo pruebas en componente <GifExpertApp/> Debe de mostrar yba lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Haciendo pruebas en componente <GifExpertApp/> Tiene que hacer match con el snap 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
